diff --git a/Makefile b/Makefile
index 7296039..1339f42 100644
--- a/Makefile
+++ b/Makefile
@@ -1,11 +1,10 @@
 # contrib/pg_audit/Makefile
 
 MODULE_big = pgaudit
-OBJS = pgaudit.o $(WIN32RES)
+OBJS = pgaudit.o
 
 EXTENSION = pgaudit
 DATA = pgaudit--18.0.sql
-PGFILEDESC = "pgAudit - An audit logging extension for PostgreSQL"
 
 REGRESS = pgaudit
 REGRESS_OPTS = --temp-config=$(top_srcdir)/contrib/pgaudit/pgaudit.conf
