diff --git a/build/config/posix/BUILD.gn b/build/config/posix/BUILD.gn
index 8312d7e..9ef65b2 100644
--- a/build/config/posix/BUILD.gn
+++ b/build/config/posix/BUILD.gn
@@ -28,7 +28,10 @@ config("runtime_library") {
   if (!is_apple && sysroot != "" && current_os != "zos") {
     # Pass the sysroot to all C compiler variants, the assembler, and linker.
     sysroot_flags = [ "--sysroot=" + rebase_path(sysroot, root_build_dir) ]
-    if (is_linux || is_chromeos) {
+    if ((is_linux || is_chromeos) &&
+        (current_cpu == "x86" || current_cpu == "x64" ||
+         current_cpu == "arm" || current_cpu == "arm64" ||
+         current_cpu == "mipsel" || current_cpu == "mips64el")) {
       # This is here so that all files get recompiled after a sysroot roll and
       # when turning the sysroot on or off. (defines are passed via the command
       # line, and build system rebuilds things when their commandline
