--- a/Modules/Setup.stdlib.in
+++ b/Modules/Setup.stdlib.in
@@ -170,9 +170,9 @@
 ############################################################################
 # Windows specific modules
 
-@MODULE__OVERLAPPED_TRUE@_overlapped overlapped.c
-@MODULE__WMI_TRUE@_wmi ../PC/_wmimodule.cpp
-@MODULE_WINSOUND_TRUE@winsound ../PC/winsound.c
+# _overlapped overlapped.c
+# _wmi ../PC/_wmimodule.cpp
+# winsound ../PC/winsound.c
 
 
 ############################################################################
@@ -193,7 +193,7 @@
 @MODULE__TESTMULTIPHASE_TRUE@_testmultiphase _testmultiphase.c
 @MODULE__TESTSINGLEPHASE_TRUE@_testsinglephase _testsinglephase.c
 @MODULE__CTYPES_TEST_TRUE@_ctypes_test _ctypes/_ctypes_test.c
-@MODULE__TESTCONSOLE_TRUE@_testconsole ../PC/_testconsole.c
+# _testconsole ../PC/_testconsole.c
 
 # Limited API template modules; must be built as shared modules.
 @MODULE_XXLIMITED_TRUE@xxlimited xxlimited.c
--- a/Modules/Setup.bootstrap.in
+++ b/Modules/Setup.bootstrap.in
@@ -8,7 +8,7 @@
 # module C APIs are used in core
 atexit atexitmodule.c
 faulthandler faulthandler.c
-@MODULE_POSIXMODULE_NAME@ posixmodule.c
+nt posixmodule.c
 _signal signalmodule.c -lws2_32
 _tracemalloc _tracemalloc.c
 _suggestions _suggestions.c
@@ -39,9 +39,9 @@
 _symtable symtablemodule.c
 
 # for systems without $HOME env, used by site._getuserbase()
-@MODULE_PWD_TRUE@pwd pwdmodule.c
+#pwd pwdmodule.c
 
 # build-in modules for windows platform:
-@MODULE_WINREG_TRUE@winreg ../PC/winreg.c
-@MODULE_MSVCRT_TRUE@msvcrt -DPy_BUILD_CORE ../PC/msvcrtmodule.c
-@MODULE__WINAPI_TRUE@_winapi _winapi.c
+winreg ../PC/winreg.c
+msvcrt -DPy_BUILD_CORE ../PC/msvcrtmodule.c
+_winapi _winapi.c
--- a/PC/errmap.h
+++ b/PC/errmap.h
@@ -1,3 +1,8 @@
+#ifdef __MINGW32__
+#  include <winsock2.h>
+#  include <windows.h>
+#endif
+
 int
 winerror_to_errno(int winerror)
 {
--- a/Objects/dictobject.c
+++ b/Objects/dictobject.c
@@ -550,7 +550,7 @@
 #if SIZEOF_LONG == SIZEOF_SIZE_T
     minsize = Py_MAX(minsize, PyDict_MINSIZE);
     return _Py_bit_length(minsize - 1);
-#elif defined(MS_WINDOWS)
+#elif defined(MS_WINDOWS) && !defined(__MINGW32__)
     // On 64bit Windows, sizeof(long) == 4. We cannot use _Py_bit_length.
     minsize = Py_MAX(minsize, PyDict_MINSIZE);
     unsigned long msb;
--- a/Python/ceval.c
+++ b/Python/ceval.c
@@ -367,6 +367,8 @@
 
 #if defined(__s390x__)
 #  define Py_C_STACK_SIZE 320000
+#elif defined(__MINGW32__)
+#  define Py_C_STACK_SIZE 4000000
 #elif defined(_WIN32)
    // Don't define Py_C_STACK_SIZE, ask the O/S
 #elif defined(__ANDROID__)
--- a/Python/ceval_gil.c
+++ b/Python/ceval_gil.c
@@ -1,4 +1,8 @@
 #include "Python.h"
+#ifdef __MINGW32__
+#  include <windows.h>
+#endif
+
 #include "pycore_ceval.h"         // _PyEval_SignalReceived()
 #include "pycore_gc.h"            // _Py_RunGC()
 #include "pycore_initconfig.h"    // _PyStatus_OK()
--- a/Python/pylifecycle.c
+++ b/Python/pylifecycle.c
@@ -1,6 +1,10 @@
 /* Python interpreter top-level routines, including init/exit */
 
 #include "Python.h"
+#ifdef __MINGW32__
+#  include <windows.h>
+#endif
+
 #include "pycore_audit.h"         // _PySys_ClearAuditHooks()
 #include "pycore_call.h"          // _PyObject_CallMethod()
 #include "pycore_ceval.h"         // _PyEval_FiniGIL()
--- a/Python/traceback.c
+++ b/Python/traceback.c
@@ -1,6 +1,10 @@
 /* Traceback implementation */
 
 #include "Python.h"
+#ifdef __MINGW32__
+#  include <windows.h>
+#endif
+
 #include "pycore_call.h"          // _PyObject_CallMethodFormat()
 #include "pycore_fileutils.h"     // _Py_BEGIN_SUPPRESS_IPH
 #include "pycore_frame.h"         // PyFrameObject
