diff --git a/Makefile.pre.in b/Makefile.pre.in
index 734cbcf..d428992 100644
--- a/Makefile.pre.in
+++ b/Makefile.pre.in
@@ -1808,7 +1808,7 @@ install: @FRAMEWORKINSTALLFIRST@ commoninstall bininstall maninstall @FRAMEWORKI
 			upgrade) ensurepip="--upgrade" ;; \
 			install|*) ensurepip="" ;; \
 		esac; \
-		$(RUNSHARED) $(PYTHON_FOR_BUILD) -m ensurepip \
+		$(RUNSHARED) _PYTHON_HOST_PLATFORM=linux-$(shell uname -m) $(PYTHON_FOR_BUILD) -m ensurepip \
 			$$ensurepip --root=$(DESTDIR)/ ; \
 	fi

@@ -1818,7 +1818,7 @@ altinstall: commoninstall
 			upgrade) ensurepip="--altinstall --upgrade" ;; \
 			install|*) ensurepip="--altinstall" ;; \
 		esac; \
-		$(RUNSHARED) $(PYTHON_FOR_BUILD) -m ensurepip \
+		$(RUNSHARED) _PYTHON_HOST_PLATFORM=linux-$(shell uname -m) $(PYTHON_FOR_BUILD) -m ensurepip \
 			$$ensurepip --root=$(DESTDIR)/ ; \
 	fi

@@ -2361,6 +2361,7 @@ endif
 sharedinstall: all
 	MSYS2_ARG_CONV_EXCL="--prefix=;--install-scripts=;--install-platlib=" \
 	$(RUNSHARED) $(PYTHON_FOR_BUILD) $(srcdir)/setup.py install \
+		--skip-build \
 	   	--prefix=$(prefix) \
 		--install-scripts=$(BINDIR) \
 		--install-platlib=$(DESTSHARED) \
diff --git a/Modules/Setup.bootstrap.in b/Modules/Setup.bootstrap.in
index 6986290..2db445a 100644
--- a/Modules/Setup.bootstrap.in
+++ b/Modules/Setup.bootstrap.in
@@ -8,7 +8,7 @@
 # module C APIs are used in core
 atexit atexitmodule.c
 faulthandler faulthandler.c
-@INITSYS@ posixmodule.c
+nt posixmodule.c
 _signal signalmodule.c -lws2_32
 _tracemalloc _tracemalloc.c
 
@@ -32,9 +32,9 @@ _stat _stat.c
 _symtable symtablemodule.c
 
 # for systems without $HOME env, used by site._getuserbase()
-@MODULE_PWD_TRUE@pwd pwdmodule.c
+#pwd pwdmodule.c
 
 # build-in modules for windows platform:
-@USE_WIN32_MODULE@winreg ../PC/winreg.c
-@MODULE_MSVCRT_TRUE@msvcrt -DPy_BUILD_CORE ../PC/msvcrtmodule.c
-@MODULE__WINAPI_TRUE@_winapi _winapi.c
+winreg ../PC/winreg.c
+msvcrt -DPy_BUILD_CORE ../PC/msvcrtmodule.c
+_winapi _winapi.c
diff --git a/PC/errmap.h b/PC/errmap.h
index a7489ab..fba7170 100644
--- a/PC/errmap.h
+++ b/PC/errmap.h
@@ -1,3 +1,8 @@
+#ifdef __MINGW32__
+#  include <winsock2.h>
+#  include <windows.h>
+#endif
+
 int
 winerror_to_errno(int winerror)
 {
