--- a/win32/GNUmakefile
+++ b/win32/GNUmakefile
@@ -1176,7 +1176,7 @@
 
 $(GLOBEXE) : perlglob.c
 ifeq ($(CCTYPE),GCC)
-	$(LINK32) $(EXTRACFLAGS) $(OPTIMIZE) $(BLINK_FLAGS) -mconsole -o $@ perlglob.c $(LIBFILES)
+	$(CC) $(EXTRACFLAGS) $(OPTIMIZE) $(BLINK_FLAGS) -mconsole -o $@ perlglob.c $(LIBFILES)
 else
 	$(CC) $(EXTRACFLAGS) $(OPTIMIZE) $(PDBOUT) -Fe$@ perlglob.c -link $(BLINK_FLAGS) \
 	setargv$(o) $(LIBFILES)
@@ -1440,7 +1440,7 @@
 
 perllib$(o)	: perllib.c perllibst.h .\perlhost.h .\vdir.h .\vmem.h
 ifeq ($(USE_IMP_SYS),define)
-	$(CC) -c -I$(FULLDIR) -I. $(CFLAGS_O) $(CXX_FLAG) $(OBJOUT_FLAG)$@ $(PDBOUT) perllib.c
+	$(CC) -c -I$(FULLDIR) -I. $(subst -std=c99,,$(CFLAGS_O)) $(CXX_FLAG) $(OBJOUT_FLAG)$@ $(PDBOUT) perllib.c
 else
 	$(CC) -c -I$(FULLDIR) -I. $(CFLAGS_O) $(OBJOUT_FLAG)$@ $(PDBOUT) perllib.c
 endif
@@ -1508,7 +1508,7 @@
 
 $(GENUUDMAP) : ..\mg_raw.h
 ifeq ($(CCTYPE),GCC)
-	$(LINK32) $(CFLAGS_O) -o..\generate_uudmap.exe ..\generate_uudmap.c \
+	$(CC) $(CFLAGS_O) -o..\generate_uudmap.exe ..\generate_uudmap.c \
 	$(BLINK_FLAGS) $(LIBFILES)
 else
 	$(CC) $(CFLAGS_O) $(PDBOUT) -Fe..\generate_uudmap.exe ..\generate_uudmap.c -link $(LIBFILES) $(BLINK_FLAGS)
@@ -1521,10 +1521,16 @@
 
 # also known as $(HAVE_COREDIR)
 .coreheaders : $(CORE_H)
-	$(XCOPY) *.h $(COREDIR)\\*.*
-	$(RCOPY) include $(COREDIR)\\*.*
-	$(XCOPY) ..\\*.h $(COREDIR)\\*.*
-	$(XCOPY) ..\\charclass_invlists.inc $(COREDIR)\\*.*
+	if not exist "$(COREDIR)" mkdir "$(COREDIR)"
+	if not exist "$(COREDIR)\arpa" mkdir "$(COREDIR)\arpa"
+	if not exist "$(COREDIR)\sys" mkdir "$(COREDIR)\sys"
+	$(XCOPY) *.h $(COREDIR)\$(NULL)
+	$(RCOPY) include $(COREDIR)\include\$(NULL)
+	$(XCOPY) include\\*.h $(COREDIR)\$(NULL)
+	$(XCOPY) include\\arpa\\*.h $(COREDIR)\arpa\$(NULL)
+	$(XCOPY) include\\sys\\*.h $(COREDIR)\sys\$(NULL)
+	$(XCOPY) ..\\*.h $(COREDIR)\$(NULL)
+	$(XCOPY) ..\\charclass_invlists.inc $(COREDIR)\$(NULL)
 	rem. > $@
 
 perlmain$(o) : runperl.c $(CONFIGPM)
--- a/win32/perlhost.h
+++ b/win32/perlhost.h
@@ -2076,8 +2076,8 @@
 const struct IPerlProc perlProc =
 {
     PerlProcAbort,
     PerlProcCrypt,
-    PerlProcExit,
-    PerlProc_Exit,
+    (LPProcExit)PerlProcExit,
+    (LPProc_Exit)PerlProc_Exit,
     PerlProcExecl,
     PerlProcExecv,
